lib/pacemaker/pcmk_stonith.c

```diff
@@ -0,0 +1,485 @@
+/*
+ * Copyright 2009-2019 the Pacemaker project contributors
+ *
+ * The version control history for this file may have further details.
+ *
+ * This source code is licensed under the GNU General Public License version 2
+ * or later (GPLv2+) WITHOUT ANY WARRANTY.
+ */
+
+#include <crm_internal.h>
+#include <crm/common/mainloop.h>
+#include <crm/common/output.h>
+#include <crm/stonith-ng.h>
+#include <crm/fencing/internal.h>
+#include <glib.h>
+#include <libxml/tree.h>
+#include <pacemaker.h>
+#include <pcmki/pcmki_stonith.h>
+
+static pcmk__supported_format_t formats[] = {
+    PCMK__SUPPORTED_FORMAT_XML,
+    { NULL, NULL, NULL }
+};
```

Comment:
This will likely be used for all public APIs, so maybe we can put it into a new pcmk_output.c along with prologue() and epilogue(). I'm sure we'll end up with libpacemaker messages to register, so that could go there too when the time comes.

Reply:
I was curious if prologue and epilogue might differ between major portions of the API.  Like, will whatever we pull out of crm_mon need something different?  At the moment I don't think so but maybe it would be instructive to lib-ify something from crm_mon and find out.

Reply:
It doesn't matter much since it's all internal, but I can't think of anything that would need to be done differently (vs additional work that could be in its own function).